experimental/traffic-portal/src/app/core/origins/detail/origin-detail.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { MatButtonHarness } from "@angular/material/button/testing";
+import { MatDialogModule } from "@angular/material/dialog";
+import { MatDialogHarness } from "@angular/material/dialog/testing";
+import { ActivatedRoute } from "@angular/router";
+import { RouterTestingModule } from "@angular/router/testing";
+import { ReplaySubject } from "rxjs";
+
+import { OriginService } from "src/app/api";
+import { APITestingModule } from "src/app/api/testing";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+import { OriginDetailComponent } from "./origin-detail.component";
+
+describe("OriginDetailComponent", () => {
+	let component: OriginDetailComponent;
+	let fixture: ComponentFixture<OriginDetailComponent>;
+	let route: ActivatedRoute;
+	let paramMap: jasmine.Spy;
+	let loader: HarnessLoader;
+	let service: OriginService;
+
+	const navSvc = jasmine.createSpyObj([], {
+		headerHidden: new ReplaySubject<boolean>(),
+		headerTitle: new ReplaySubject<string>(),
+	});
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
+			declarations: [OriginDetailComponent],
+			imports: [APITestingModule, RouterTestingModule, MatDialogModule],
+			providers: [{provide: NavigationService, useValue: navSvc}],
+		}).compileComponents();
+
+		route = TestBed.inject(ActivatedRoute);
+		paramMap = spyOn(route.snapshot.paramMap, "get");
+		paramMap.and.returnValue(null);
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		service = TestBed.inject(OriginService);
+		loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+		await fixture.whenStable();
+	});
+
+	it("should create", () => {
+		expect(component).toBeTruthy();
+	});
+
+	it("new origin", async () => {
+		paramMap.and.returnValue("new");
+
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		await fixture.whenStable();
+		expect(paramMap).toHaveBeenCalled();
+		expect(component.origin).not.toBeNull();
+		expect(component.origin.name).toBe("");
+		expect(component.new).toBeTrue();
+	});
+
+	it("existing origin", async () => {
+		const id = 1;
+		paramMap.and.returnValue(id);
+		const origin = await service.getOrigins(id);
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		await fixture.whenStable();
+		expect(paramMap).toHaveBeenCalled();
+		expect(component.origin).not.toBeNull();
+		expect(component.origin.name).toBe(origin.name);
+		expect(component.new).toBeFalse();
+	});
+
+	it("deletes existing Origins", async () => {
+		const spy = spyOn(service, "deleteOrigin").and.callThrough();
+		let cgs = await service.getOrigins();
+		const initialLength = cgs.length;
+		if (initialLength < 1) {
+			return fail("need at least one Cache Group");
```

Comment:
You actually just need at least one _Origin_

Also, names of things like `cgs` are clearly stale copypasta from the Cache Group details page spec - should probably be e.g. `orgs`.